cf-editor-rte-toolbar-configurable-buttons/src/aem-cf-editor-1/actions/getCFModelConfig/index.js

```diff
@@ -0,0 +1,90 @@
+ *   - The two steps above imply that every client knowing the URL to this deployed action will be able to invoke it without any authentication and authorization checks against Adobe Identity Management System
+ *   - Make sure to validate these changes against your security requirements before deploying the action
+ */
+
+
+const fetch = require('node-fetch')
+const { Core } = require('@adobe/aio-sdk')
+const { errorResponse, getBearerToken, stringParameters, checkMissingRequestInputs } = require('../utils')
+
+// main function that will be executed by Adobe I/O Runtime
+async function main (params) {
+  // create a Logger
+  const logger = Core.Logger('main', { level: params.LOG_LEVEL || 'info' })
+
+  try {
+    // 'info' is the default level if not set
+    logger.info('Calling the main action')
+
+    // log parameters, only if params.LOG_LEVEL === 'debug'
+    logger.debug(stringParameters(params))
+
+    // check for missing request input parameters and headers
+    const requiredParams = [/* add required params */]
+    const requiredHeaders = ['Authorization']
+    // const errorMessage = checkMissingRequestInputs(params, requiredParams, requiredHeaders)
+    // if (errorMessage) {
+    //   // return and log client errors
+    //   return errorResponse(400, errorMessage, logger)
+    // }
+
+    // extract the user Bearer token from the Authorization header
+    const token = getBearerToken(params)
+
+    // replace this with the api you want to access
+    // const apiEndpoint = 'https://adobeioruntime.net/api/v1'
+    //
+    // // @todo fetch data about extra config for the model fields
+    // const res = await fetch(apiEndpoint)
+    // if (!res.ok) {
+    //   throw new Error('request to ' + apiEndpoint + ' failed with status code ' + res.status)
+    // }
+    // const content = await res.json()
+
+    //allowed buttons
+    const inputString = "aligncenter | alignjustify | alignleft | alignnone | alignright | blockquote | backcolor | bold | copy | cut | fontselect | fontsizeselect | forecolor | formatselect | h1 | h2 | h3 | h4 | h5 | h6 | indent | italic | language | lineheight | newdocument | outdent | paste | redo | remove | removeformat | selectall | strikethrough | styleselect | subscript | superscript | underline | undo | visualaid | pastetext | link | openlink | unlink | table | tablecellprops | tablecopyrow | tablecutrow | tabledelete | tabledeletecol | tabledeleterow | tableinsertdialog | tableinsertcolafter | tableinsertcolbefore | tableinsertrowafter | tableinsertrowbefore | tablemergecells | tablepasterowafter | tablepasterowbefore | tableprops | tablerowprops | tablesplitcells | tableclass | tablecellclass | tablecellvalign | tablecellborderwidth | tablecellborderstyle | tablecaption | tablecellbackgroundcolor | tablecellbordercolor | tablerowheader | tablecolheader | code | fullscreen | bullist | numlist | charmap | preview | searchreplace | visualblocks | insertdatetime | media | anchor";
```

Comment:
Yes, that's exactly what I had in mind. To have a list in a parsable format. The only problem with your implementation is due to our current API. We can only remove and state what toolbar we actually want but this looks great.

I would assume the string coming from the CFM metadata would be JSON, but that's a minor detail.